notification-server/src/processor.rs

```diff
@@ -101,3 +102,317 @@ pub async fn process(
+        rng.fill(&mut address);
+        address
+    }
+
+    fn split_u64_to_random_vec(
+        mut value: u64,
+        max_elements: usize,
+        g: &mut Gen,
+    ) -> Vec<(Timestamp, Amount)> {
+        let mut rng = thread_rng();
+        let mut result = Vec::new();
+
+        while value > 0 && result.len() < max_elements {
+            let part = rng.gen_range(0..=value);
+            result.push((u64::arbitrary(g).into(), Amount { micro_ccd: part }));
+            value -= part;
+        }
+
+        if value > 0 {
+            result.push((u64::arbitrary(g).into(), Amount { micro_ccd: value }));
+        }
+        result
+    }
+
+    fn _get_random_cis2_event(g: &mut Gen, amount: Amount, address: AccountAddress) -> Event {
+        let token_id = TokenId::new("wCCD".as_bytes().to_vec()).unwrap();
+        let events = vec![
+            Event::Transfer {
+                token_id: token_id.clone(),
+                amount:   TokenAmount(amount.micro_ccd.into()),
+                from:     Account(AccountAddress(random_account_address())),
+                to:       Account(address),
+            },
+            Event::Mint {
+                token_id: token_id.clone(),
+                amount:   TokenAmount(amount.micro_ccd.into()),
+                owner:    Account(address),
+            },
+            Event::Burn {
+                token_id: token_id.clone(),
+                amount:   TokenAmount(amount.micro_ccd.into()),
+                owner:    Account(address),
+            },
+        ];
+        g.choose(&events).unwrap().clone()
+    }
+
+    #[derive(Clone, Debug)]
+    struct EmittingBlockItemSummaryPair(pub BlockItemSummary, pub NotificationInformation);
+
+    impl Arbitrary for EmittingBlockItemSummaryPair {
+        fn arbitrary(g: &mut Gen) -> Self {
+            let amount = Amount {
+                micro_ccd: u64::arbitrary(g),
+            };
+            let receiver_address = AccountAddress(random_account_address());
+
+            let details = |effects| AccountTransactionDetails {
+                cost: amount.clone(),
+                effects,
+                sender: receiver_address.clone(),
+            };
+            let effects = vec![
+                AccountTransactionEffects::AccountTransfer {
+                    amount: amount.clone(),
+                    to:     receiver_address.clone(),
+                },
+                AccountTransactionEffects::AccountTransferWithMemo {
+                    amount: amount.clone(),
+                    to:     receiver_address.clone(),
+                    memo:   random_memo(),
+                },
+                AccountTransactionEffects::TransferredWithSchedule {
+                    to:     receiver_address.clone(),
+                    amount: split_u64_to_random_vec(amount.clone().micro_ccd, 2, g),
+                },
+                AccountTransactionEffects::TransferredWithScheduleAndMemo {
```

Comment:
Is the fact that these are missing recorded somewhere (e.g. an issue)? otherwise it might make sense to add an in-code comment here that these are missing.

Reply:
More a comment to the reviewer

Reply:
I will add an in-code comment. And yes it is an issue, however I also need to address CIS2 tokens more

https://concordium.atlassian.net/wiki/spaces/EN/pages/1502019590/Message+data+payload

A lot of changes has to be made on CIS-2 tokens